torch/testing/_internal/common_distributed.py

```diff
@@ -359,9 +360,32 @@ def _spawn_processes(self):
         proc = torch.multiprocessing.get_context("spawn").Process
         self._start_processes(proc)
 
+    class Event(Enum):
+        GET_TRACEBACK = 1
+
+    @staticmethod
+    def _event_listener(pipe):
+        while True:
+            event = pipe.recv()
+            if event == MultiProcessTestCase.Event.GET_TRACEBACK:
+                # Return traceback to the parent process.
+                with tempfile.NamedTemporaryFile(mode='r+') as tmp_file:
+                    faulthandler.dump_traceback(tmp_file)
+                    # Flush buffers and seek to read from the beginning
+                    tmp_file.flush()
+                    tmp_file.seek(0)
+                    pipe.send(tmp_file.read())
```

Comment:
Can't we exit after this send, since the subprocess shouldn't get multiple dump traceback requests? If so, we shouldn't need the `while True` either right, since `recv()` will block until we get a message?

Reply:
I just kept this pattern so this event_listener can get extended in the future with maybe other things we want it to do.